10.1.0/startup/generateStartupCommand.js

```diff
@@ -1,78 +1,180 @@
+            if (typeof json == 'object' && typeof json.scripts == 'object' && typeof json.scripts.start == 'string') {
+                console.log("Found scripts.start in " + packageJsonPath);
+                fs.writeFileSync(PACKAGE_JSON_FLAG, "TRUE"); // set PACKAGE_JSON_FLAG for remote debugging
+                if (packageJsonPath != "package.json") {
+                    var packageJsonDir = packageJsonPath.substring(0, packageJsonPath.indexOf("package.json")-1);
+                    return "npm --prefix=" + packageJsonDir + " start";
+                } else {
+                    return 'npm start';
+                }
+            }
+        }
+    }
+    
+    // look for process.json or *.json
+    if (userStartupCommand.endsWith(".json") || !userStartupCommand) {
+        var processJsonPath = "";
+        if (!userStartupCommand) {
+            processJsonPath = "/home/site/wwwroot/process.json";
+        } else {
+            processJsonPath = userStartupCommand;
+        }
+        if (fs.existsSync(processJsonPath)) {
+            var json = JSON.parse(fs.readFileSync(processJsonPath, 'utf8'))
+            console.log("Found script in " + processJsonPath)
+            fs.writeFileSync(PROCESS_JSON_FLAG, "TRUE"); // set PROCESS_JSON_FLAG for remote debugging
+            if (process.env.APPSVC_REMOTE_DEBUGGING == "TRUE") {
+                if (typeof json == 'object' && typeof json.script == 'string') {
+                    var nodeFile = json.script; // run the script directly with node
+                    if (process.env.APPSVC_REMOTE_DEBUGGING_BREAK == "TRUE") {	
+                        return "node --inspect-brk=0.0.0.0:" + process.env.APPSVC_TUNNEL_PORT + " " + nodeFile;
+                    } else {
+                        return "node --inspect=0.0.0.0:" + process.env.APPSVC_TUNNEL_PORT + " " + nodeFile;
+                    }
+                } else {
+                    // did not find script tag
+                    return "pm2 start " + processJsonPath + " --no-daemon";
+                }
+            } else {
+                // Run with pm2
+                return "pm2 start " + processJsonPath + " --no-daemon";
+            }
+        }
+    }
+
+    // look for *.config.js
+    if (userStartupCommand.endsWith(".config.js") || !userStartupCommand) {
+        var configJsPath = "";
+        if (!userStartupCommand) {
+            configJsPath = "/home/site/wwwroot/ecosystem.config.js";
+        } else {
+            configJsPath = userStartupCommand;
+        }
+        if (fs.existsSync(configJsPath)) {
+            console.log("Found script in " + configJsPath)
+            fs.writeFileSync(PROCESS_JSON_FLAG, "TRUE"); // set PROCESS_JSON_FLAG for remote debugging
+
+            if (process.env.APPSVC_REMOTE_DEBUGGING == "TRUE") {
+                // debugging not yet supported
+                return "pm2 start " + configJsPath + " --no-daemon";
```

Comment:
Is there a way for customers to specify the `env` that they want run or the app they want to run using the `--only` variables?